monopod/pkg/commands/scaffold.go

```diff
@@ -72,47 +78,86 @@ func (o scaffoldOptions) runScaffold(ctx context.Context) error {
 		return err
 	}
 
-	modroot := filepath.Join(o.OutputPath, "images", o.ModuleName)
+	if err := o.mkImageDir(); err != nil {
+		return err
+	}
 
-	if err := os.MkdirAll(modroot, os.ModePerm); err != nil {
-		return fmt.Errorf("unable to generate target folder images/%s: %w", o.ModuleName, err)
+	if err := filepath.Walk("images/TEMPLATE", o.walk); err != nil {
+		return err
 	}
 
-	if err := filepath.Walk("images/TEMPLATE", func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
+	// create readme hcl metadata file
+	nr := newReadmeImpl{
+		Readme:      completeReadme{Name: o.PackageName},
+		renderedHCl: new(bytes.Buffer),
+	}
+	if err := nr.render(); err != nil {
+		return err
+	}
+	if err := nr.write(); err != nil {
+		return err
+	}
 
-		if info.IsDir() {
-			return nil
-		}
-		repl := strings.Replace(path, "TEMPLATE", o.ModuleName, 1) // Replacing TEMPLATE with the module name
-		if err := os.MkdirAll(filepath.Dir(repl), os.ModePerm); err != nil {
-			return err
-		}
+	// render the readme's markdown
+	rr := NewReadmeRenderer(o.PackageName)
+	rr.Readme = &nr.Readme
+	if err := rr.render(); err != nil {
+		return err
+	}
+	if err := rr.write(); err != nil {
+		return err
+	}
 
-		if strings.HasSuffix(repl, ".tpl") {
-			f, err := os.Create(strings.TrimSuffix(repl, ".tpl"))
-			if err != nil {
-				panic(err.Error())
-			}
-			defer f.Close()
-			log.Println("writing file", f.Name())
-			return template.Must(template.ParseFiles(path)).Execute(f, o)
-		}
+	// read, render, then write main.tf
+	if err := o.readMainTf(); err != nil {
+		return err
+	}
+	if err := o.addModuleToMainTf(); err != nil {
+		return err
+	}
+	return o.writeMainTf()
+}
+
+func (o *scaffoldOptions) walk(path string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
 
-		// Just copy the file without templating.
-		b, err := os.ReadFile(path)
+	if info.IsDir() {
+		return nil
+	}
+	repl := strings.Replace(path, "TEMPLATE", o.ModuleName, 1) // Replacing TEMPLATE with the module name
+	if err := os.MkdirAll(filepath.Dir(repl), os.ModePerm); err != nil {
+		return err
+	}
+
+	if strings.HasSuffix(repl, ".tpl") {
+		f, err := os.Create(strings.TrimSuffix(repl, ".tpl"))
 		if err != nil {
-			return err
+			panic(err.Error())
```

Comment:
```suggestion
			return err
```

Reply:
No idea how/why that made it in, good 👁️